crates/rpc/src/module.rs

```diff
@@ -9,6 +9,19 @@ use crate::error::RpcError;
 /// hosted at a single endpoint.
 pub struct Module(jsonrpsee::RpcModule<RpcContext>);
 
+/// Splits the internal RPC method name, which is in the form of
+/// `apiVersion_proper_methodName` into two separate strings:
+/// - `apiVersion`, which is the RPC API semver string
+/// - `proper_methodName`, which is the proper name of the RPC method,
+/// including the namespace (`starknet` or `pathfinder`), as described
+/// in the spec for that particular API version
+pub(crate) fn split_version_prefix(method: &str) -> (String, String) {
```

Comment:
Probably doesn't require allocation. But not sure if it matters, since its just used at startup (I think)?
```suggestion
pub(crate) fn split_version_prefix(method: &str) -> (&str, &str) {
```

Reply:
I'm afraid we have to use an owned string here. The problem is lifetimes. We get a non-`'static` `&str` from the logger api and the metrics api requires us either to provide a `&' static str` or an owned string :cry: .